app/helpers/automated_tests_helper.rb

```diff
@@ -136,96 +136,59 @@ def process_test_form(assignment, params)
     # Retrieve all test scripts
     testscripts = params[:test_scripts_attributes]
 
-    # First check for duplicate script names in test scripts
-    unless testscripts.nil?
-      testscripts.values.each do |tfile|
-        if tfile['script_name'].respond_to?(:original_filename)
-          fname = tfile['script_name'].original_filename
-          # If this is a duplicate script name, raise error and return
-          if !file_name_array.include?(fname)
-            file_name_array << fname
-          else
-            raise I18n.t('automated_tests.duplicate_filename') + fname
-          end
-        end
-      end
-    end
-
     # Retrieve all test support files
     testsupporters = params[:test_support_files_attributes]
 
-    # Now check for duplicate file names in test support files
-    unless testsupporters.nil?
-      testsupporters.values.each do |tfile|
-        if tfile['file_name'].respond_to?(:original_filename)
-          fname = tfile['file_name'].original_filename
-          # If this is a duplicate file name, raise error and return
-          if filename_array.include?(fname)
-            raise I18n.t('automated_tests.duplicate_filename') + fname
-          else
-            filename_array << fname
-          end
-        end
-      end
-    end
-
-    # Filter out script files that need to be created and updated
+    # Create/Update test scripts
     unless testscripts.nil?
-      testscripts.each_key do |key|
+      testscripts.each do |file_num, file|
+        updated_script_files[file_num] = {}
 
-        tfile = testscripts[key]
+        # Empty file submission, skip
+        next if testscripts[file_num][:script_name].nil?
```

Comment:
@WilsonChiang This is going to be `nil` now for new test scripts.

Also, the new key (new_scripts or whatever) is being sent to the server, but it's being filtered out in the controller (see the "assignment_params") before this point. So why don't you add an extra argument to this function and actually pass in the "new_file" part of the params here, so that you can use it.